src/jit/lsraarm64.cpp

```diff
@@ -816,20 +816,38 @@ void LinearScan::TreeNodeInfoInitSIMD(GenTreeSIMD* simdTree)
 
         case SIMDIntrinsicGetItem:
             // We have an object and an item, which may be contained.
-            info->srcCount = simdTree->gtGetOp2()->isContained() ? 1 : 2;
+            info->srcCount = (simdTree->gtGetOp2()->isContained() ? 1 : 2);
+
+            op1 = simdTree->gtGetOp1();
 
-            if (!simdTree->gtGetOp2()->IsCnsIntOrI())
+            if (op1->isContained())
             {
-                // If the index is not a constant, we will need a general purpose register
-                info->internalIntCount = 1;
+                // Although GT_IND of TYP_SIMD12 reserves an internal register for reading 4 and 8 bytes from memory
+                // and assembling them into target reg, it is not required in this case.
+                op1->gtLsraInfo.internalIntCount   = 0;
+                op1->gtLsraInfo.internalFloatCount = 0;
+                info->srcCount -= 1;
+                info->srcCount += GetOperandSourceCount(op1);
+            }
 
-                // If the index is not a constant, we will use the SIMD temp location to store the vector.
-                compiler->getSIMDInitTempVarNum();
+            if (!simdTree->gtGetOp2()->IsCnsIntOrI() &&
+                (!simdTree->gtGetOp1()->isContained() || simdTree->gtGetOp1()->OperIsLocal()))
```

Comment:
Minor point, but since you have `op1`, you should use it here, otherwise it's not obvious to the reader that `simdTree->gtGetOp1()` is the same as `op1`, and the condition `simdTree->gtGetOp1()->isContained()` is the same as the condition `op1->isContained()` above.